pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -173,29 +173,34 @@ func (p *Planner) applyToMachineCondition(clusterPlan *plan.Plan, machineNames [
 		}
 
 		if !condition.Cond(capi.InfrastructureReadyCondition).IsTrue(machine) {
-			// Don't wait for CustomMachines to be ready because the infrastructure should be ready.
-			// The CustomMachine is waiting for the providerID to be set which won't happen until the cluster is bootstrapped.
-			if clusterPlan.Machines[machineName].Spec.InfrastructureRef.Kind != "CustomMachine" {
-				waiting = true
-				continue
-			}
+			waiting = true
+			continue
+		}
+
+		// Once a machine is provisioned (i.e. the first time it is bootstrapped and joined to the cluster),
+		// the machine should not go into a provisioning state again. Instead, updating is used.
+		cond := rke2.Provisioned
+		if cond.IsTrue(machine) {
```

Comment:
I'm not a fan of changing the conditions we're setting based on other conditions.

Reply:
@Oats87 Do you have another suggestion?

Reply:
If the goal is to fix a superficial issue where the user may see "provisioning" on a machine in the UI, then I would lean to fixing this at the UI level. 

My understanding is that the requirement here is entirely to solve a superficial issue where a machine says "provisioning". If that is the case, then maybe we just change the condition entirely to `Reconciled` or something along those lines. 

I see no benefit of this change to the backend other than introducing potential bugs/confusion as we copy/trickle conditions around all over the place.

Reply:
Using Reconciled instead.